sb_vision/distance_finding.py

```diff
@@ -0,0 +1,105 @@
+import functools
+from pathlib import Path
+from typing import Tuple, Dict
+
+import cv2
+from lxml import etree
+import re
+import numpy as np
+
+
+def _get_text(element: etree.Element):
+    """parse an xml tag with space-separated variables"""
+    text = []
+    for e in element.itertext():
+        if e.strip():
+            text += re.split('\s+', re.sub('\n', '', e.strip()))
+    return text
+
+
+def get_calibration(file_name: Path) -> Dict:
+    """
+    Parse a calibration xml generated by the camera calibration tool
+    (see https://docs.opencv.org/3.4.0/d7/d21/tutorial_interactive_calibration.html)
+    :param file_name: name of xml file to parse
+    :return: calibrations from the file in a dictionary
+    """
+    calibrations = {}
+    with file_name.open() as file:
+        tree = etree.parse(file)
+        root = tree.getroot()
+        for element in root:
+            if 'type_id' in element.attrib and element.attrib[
+                 'type_id'] == 'opencv-matrix':
+                rows, cols = int(element.find('rows').text), int(
+                    element.find('cols').text)
+                data_type = element.find('dt').text
+                values = _get_text(element.find('data'))
+                if data_type == 'd':  # doubles
+                    data = np.reshape(np.array([float(v) for v in values]),
+                                      (rows, cols))
+                else:
+                    raise ValueError('Invalid data type in xml')
+            # Integer tag names
+            elif element.tag in ['framesCount', 'cameraResolution']:
+                values = _get_text(element)
+                data = [int(v) for v in values]
+            elif element.tag in ['avg_reprojection_error']:
+                values = _get_text(element)
+                data = [float(v) for v in values]
+            else:
+                data = _get_text(element)
+            calibrations[element.tag] = data
+    return calibrations
+
+
+@functools.lru_cache()
+def load_camera_calibrations(file_name: Path) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
Note that typing of numpy types doesn't actually work -- they're treated as `Any`, so if we can avoid using their array types and instead use built-in types that might be better.